apps/vue-showcase/src/app/plugins/vue-route-guard.ts

```diff
@@ -0,0 +1,30 @@
+import { setupGuard } from '@this-dot/vue-route-guard';
+import * as vue from 'vue';
+import { Router } from 'vue-router';
+
+export default (app: vue.App, router: Router) => {
+  app.use(
+    setupGuard({
+      router: router,
+      token: {
+        name: 'VUEROUTEGUARD-TOKEN',
+      },
+      redirect: {
+        noAuthentication: '/route-guard/login',
```

Comment:
Once again, I'd remove the `/route-guard` part in the route simply because there's no reason to expose technical details to the public.
Instead routes could be :
- `auth/login` or `/login`
- `/forbidden` or `/no-permission` or `/error/no-permission`

Reply:
since you know it's a route-guard and it also makes us of a method `setUpGuard`, so instead of adding the route-guard to the route, you could have a prop called routeGuage(boolean) for any route which will be either true or false. This will allow for Abstraction.